kartothek/io_components/metapartition.py

```diff
@@ -1696,15 +1694,15 @@ def parse_input_to_metapartition(
     return mp
 
 
-def _ensure_valid_indices(secondary_indices, mp_indices):
-    if not secondary_indices:
-        if mp_indices:
-            raise ValueError(
-                "Incorrect indices provided for dataset.\n"
-                f"Expected index columns: {secondary_indices}"
-                f"Provided index: {mp_indices}"
-            )
-    else:
+def _ensure_valid_indices(mp_indices, secondary_indices=None, data=None):
```

Comment:
Can you move this function to `io_components/utils.py`? The behavior is closely entangled to `_ensure_compatible_indices` and should keep them close / refactor it to be less brittle

Reply:
Done. 
Have added a TODO to do refactoring also.